main.nf

```diff
@@ -711,6 +712,8 @@ process sample_correlation {
 
     script: // This script is bundled with the pipeline, in NGI-RNAseq/bin/
     def rlocation = params.rlocation ?: ''
+    prefix = reads[0].toString() - ~/(_R1)?(_trimmed)?(_val_1)?(\.fq)?(\.fastq)?(\.gz)?P\d+(_\d+)?$/
```

Comment:
Isn't this _removing_ the `P1234` number? The `reads[0].toString() - 'xxx'` removes `'xxx'` from the string. So by adding `P\d+(_\d+)?` to the end of the regex, I think you're removing that from the basename?